python/lsst/daf/persistence/butler.py

```diff
@@ -433,8 +451,18 @@ def _read(self, pythonType, location):
         return results
 
     def __reduce__(self):
-        return (_unreduce, (self.mapper,))
+        return (_unreduce, (self.mapper, self.datasetTypeAliasDict))
+
+    def __resolveDatasetTypeAlias(self, type):
```

Comment:
You should generally use a single underscore to indicate a "private/protected" method, not a double underscore (which, among other things, prevents overriding in a subclass).